bundles/org.palladiosimulator.dataflow.confidentiality.analysis/src/org/palladiosimulator/dataflow/confidentiality/analysis/resource/PCMURIResourceProvider.java

```diff
@@ -102,4 +108,74 @@ private boolean isTargetInResource(EClass targetType, Resource resource) {
         }
         return false;
     }
+	
+	/**
+	 * Finds an element that fulfills the given condition in a given resource
+	 * @param condition Condition the element should fulfill
+	 * @param resource Resource that should be searched
+	 * @return Returns the first entity, that fulfills the condition. If none are found, the method returns null
+	 */
+	private Entity findInResource(Predicate<Entity> condition, Resource resource) {
+		if (resource == null) {
```

Comment:
I would opt for Optional (pun intended) rather than returning null. Finding an element is always a type of function that could return "not found", but better in the form of `Optional.empty` than null. The same is true for all other newly added functions